tf2onnx/optimizer/back_to_back_optimizer.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+"""Back_To_Back Optimizer.
+   Collapse consecutive nodes into 1 node if possible.
+"""
+
+from __future__ import unicode_literals
+
+from .optimizer_base import GraphOptimizerBase
+
+# pylint: disable=logging-not-lazy,unused-argument,missing-docstring,unused-variable,arguments-differ
+
+_func_map = {}
+
+
+def _register_func(op_type):
+    def _internal_fun(func):
+        _func_map[op_type] = func
+        return func
+
+    return _internal_fun
+
+
+class BackToBackOptimizer(GraphOptimizerBase):
+    """Remove back-to-back nodes e.g. 'Cast'
+    """
+
+    def __init__(self):  # pylint: disable=useless-super-delegation
+        super(BackToBackOptimizer, self).__init__()
+
+    def _optimize(self, graph):
+        return self._apply_optimization(graph, self._optimize_at_current_graph_level)
+
+    def _optimize_at_current_graph_level(self, g):
+        for optype, handler in _func_map.items():
+            nodes = [n for n in g.get_nodes() if n.type == optype]
+            # find consumer list for each cast node
+            has_dependencies = set()
+            consumer_node_ids = {n.output[0]: [] for n in nodes}
```

Comment:
There is a method on the Graph class called find_consumers() that we use instead of doing that manual.

Reply:
I noticed the method you  mentioned. The `find_output_consumers()` method find all nodes that consume a particular output, however the consumers can be different op_type (i.e. doesn't not filter by op-type,). Also, it makes a pass over all the nodes. It seemed a bit more efficient this way. Let me know thoughts.